tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousTest.java

```diff
@@ -65,34 +68,20 @@ public static WebArchive deploy() {
      */
     @Test
     public void testAsyncIsNotFinished() {
-        Future<Connection> future = null;
-        try {
-            future = client.service();
-        }
-        catch (InterruptedException e) {
-            throw new AssertionError("testAsync: unexpected InterruptedException calling service");
-        }
-
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        Future<Connection> future = client.service(waitingFuture);
         Assert.assertFalse(future.isDone());
-
     }
 
     /**
      * Test that the future returned by calling an asynchronous method is done if called after waiting enough time to end the operation
      */
     @Test
     public void testAsyncIsFinished() {
-        Future<Connection> future = null;
-        try {
-            future = client.service();
-            Thread.sleep(1500);
-        }
-        catch (InterruptedException e) {
-            throw new AssertionError("testAsync: unexpected InterruptedException calling service");
-        }
-
-        Assert.assertTrue(future.isDone());
-
+        CompletableFuture<Void> waitingFuture = newWaitingFuture();
+        Future<Connection> future = client.service(waitingFuture);
+        await().atLeast(1000, TimeUnit.MILLISECONDS).atMost(2000, TimeUnit.MILLISECONDS)
```

Comment:
@Ladicek suggested not to wait for 1s ` .atLeast(1000, TimeUnit.MILLISECONDS)` is to be removed.

Reply:
done